k9mail/src/main/java/com/fsck/k9/activity/setup/AccountSetupCheckSettings.java

```diff
@@ -509,6 +516,43 @@ private void checkIncoming() throws MessagingException {
                     .synchronizeMailbox(account, account.getInboxFolder(), null, null);
         }
 
+        private void createSpecialLocalFolders(CheckDirection direction) throws MessagingException {
+            if (direction != CheckDirection.INCOMING) {
+                return;
+            }
+
+            LocalStore localStore = account.getLocalStore();
+            createLocalFolder(localStore, Account.OUTBOX, getString(R.string.special_mailbox_name_outbox));
+
+            if  (!account.getStoreUri().startsWith("pop3")) {
```

Comment:
This seems hacky. Maybe let the remote store itself decide whether it wants these set up?

Reply:
It is hacky. The trouble is `Pop3Store` isn't the right place for it, either. It's only concerned about the server communication.
I plan to introduce an abstraction layer I called `Backend`. A backend will only expose high level functions to K-9 Mail (setup, check mail, mark message as read, …). It's then up to the Backend to implement them. The implementation will also be the place for setup code like this.